src/lib.rs

```diff
@@ -1746,6 +1812,43 @@ impl<K: KVStore + Sync + Send + 'static> Drop for Node<K> {
 	}
 }
 
+/// Represents the status of the [`Node`].
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct NodeStatus {
+	/// Indicates whether the [`Node`] is running.
+	pub is_running: bool,
+	/// Indicates whether the [`Node`] is listening for incoming connections on the addresses
+	/// configured via [`Config::listening_addresses`].
+	pub is_listening: bool,
+	/// The best block to which our Lightning wallet is currently synced.
+	pub current_best_block: BestBlock,
+	/// The timestamp, in seconds since start of the UNIX epoch, when we last successfully synced
+	/// our Lightning wallet to the chain tip.
+	///
+	/// Will be `None` if the wallet hasn't been synced since the [`Node`] was initialized.
+	pub latest_wallet_sync_timestamp: Option<u64>,
+	/// The timestamp, in seconds since start of the UNIX epoch, when we last successfully synced
+	/// our on-chain wallet to the chain tip.
+	///
+	/// Will be `None` if the wallet hasn't been synced since the [`Node`] was initialized.
+	pub latest_onchain_wallet_sync_timestamp: Option<u64>,
+	/// The timestamp, in seconds since start of the UNIX epoch, when we last successfully update
+	/// our fee rate cache.
+	///
+	/// Will be `None` if the cache hasn't been updated since the [`Node`] was initialized.
+	pub latest_fee_rate_cache_update_timestamp: Option<u64>,
+	/// The timestamp, in seconds since start of the UNIX epoch, when the last rapid gossip sync
+	/// (RGS) snapshot we successfully applied was generated.
+	///
+	/// Will be `None` if RGS isn't configured or the snapshot hasn't been updated since the [`Node`] was initialized.
+	pub latest_rgs_snapshot_timestamp: Option<u64>,
+	/// The timestamp, in seconds since start of the UNIX epoch, when we last broadcasted a node
+	/// announcement.
+	///
+	/// Will be `None` if we have no public channels or we haven't broadcasted since the [`Node`] was initialized.
+	pub latest_node_announcement_broadcast_timestamp: Option<u64>,
```

Comment:
Would it make sense to use `Duration` for these?

Reply:
Hmm... `Duration` is a timestamp when interpreted as relative to the Unix epoch. It just gives you more precision than `u64` seconds and may be more useful interface to work with. But fair enough regarding bindings.